src/language_features/completion.rs

```diff
@@ -41,6 +41,7 @@ pub fn editor_completion(
     let unescape_markdown_re = Regex::new(r"\\(?P<c>.)").unwrap();
     let maxlen = items.iter().map(|x| x.label.len()).max().unwrap_or(0);
     let escape_bar = |s: &str| s.replace("|", r"\|");
+    let snippet_prefix_re = Regex::new(r"^[^\[\(<\n\$]+").unwrap();
```

Comment:
Are all other characters safe? Might worth restricting to word characters like `r"^\w+"` or even just ASCII alphanumeric like `r"^[[:alnum:]]+"`. I'm not sure because I don't fully understand what's fine to provide Kakoune.

Reply:
I think it should be fine, as long as we regex::escape the stuff that's used to select it afterwards. I did realize that I accidentally regex::escaped both though...

Reply:
I now regex escape the correct one. I believe i should in theory be able to send anything that's not a newline, but i thought something like this would make sense. It works even for snippets like `sizeof...(${1:parameter-pack})` now (the dots being handled correctly). I can't seem to find any other example where it breaks in rust or C++ at least...

I'd keep it like this, word characters only might break in languages that allow snake-case or snippets with operators etc.